app/actors/hyrax/actors/apply_permission_template_actor.rb

```diff
@@ -13,8 +13,23 @@ def create(env)
       private
 
         def add_edit_users(env)
-          return if env.attributes[:source_id].blank?
-          template = Hyrax::PermissionTemplate.find_by!(source_id: env.attributes[:source_id])
+          add_admin_set_participants(env)
+          add_collection_participants(env)
+        end
+
+        def add_admin_set_participants(env)
+          return if env.attributes[:admin_set_id].blank?
+          template = Hyrax::PermissionTemplate.find_by!(source_id: env.attributes[:admin_set_id])
```

Comment:
Should you also add the `source_type` to the `find_by!` (e.g. `find_by!(source_type: 'AdminSet', source_id: env.attributes[:admin_set_id])`?

Reply:
I am torn about `source_type`.  We don't use it anywhere.  It is nice having it in the table for debugging, but ultimately unnecessary.  All ids are unique.  So the source_id itself is sufficient for determining whether the source is an admin set or a collection.

Reply:
We do make use of `source_type` in this one spot:
https://github.com/samvera/hyrax/blob/collections-%231545-participants_in_new_work/app/controllers/hyrax/admin/permission_template_accesses_controller.rb#L31-L39

Reply:
And I just found a place in abilities where it will be used.

@jeremyf For this code, adding `source_type:` to the `find_by!` query would be a validation.  It won't effect what gets returned, but would throw an exception if the source_type was not admin_set.  Do you still think it is needed?

Reply:
I went with @jeremyf suggestion, since I haven't heard back, and it doesn't change the impact of the code.